pkg/proc/fncall.go

```diff
@@ -392,12 +422,12 @@ func evalFunctionCall(scope *EvalScope, node *ast.CallExpr) (*Variable, error) {
 
 		// adjust the value of registers inside scope
 		pcreg, bpreg, spreg := scope.Regs.Reg(scope.Regs.PCRegNum), scope.Regs.Reg(scope.Regs.BPRegNum), scope.Regs.Reg(scope.Regs.SPRegNum)
+		scope.Regs.AddReg(scope.Regs.BPRegNum, bpreg)
```

Comment:
This change doesn't make sense, you are setting bpreg just before the call to ClearRegisters which deletes it which is equivalent to not setting it at all. Also I think we should set it, why wouldn't we?

Reply:
Sorry that was a messup on my part while making the changes have rectified it now